includes/admin/class-wp-job-manager-admin-notices.php

```diff
@@ -317,16 +332,40 @@ private static function generate_notice_from_updates( $updates ) {
 	 * Renders a notice.
 	 *
 	 * @param array $notice See `generate_notice_from_updates` for format.
+	 * @param string $notice_id  Unique identifier for the notice.
 	 * @return void
 	 */
-	private static function render_notice( $notice ) {
-		// TODO Handle different levels of notices.
-		echo '<div class="notice wpjm-admin-notice wpjm-admin-notice--info">';
+	private static function render_notice( $notice_id, $notice ) {
+		if ( empty( $notice['actions'] ) || ! is_array( $notice['actions'] ) ) {
+			$notice['actions'] = [];
+		}
+
+		$notice_class = '';
+		if ( ! empty( $notice['style'] ) ) {
+			$notice_class = 'wpjm-notice-' . $notice['style'];
+		}
+
+		$is_dismissible       = $notice['dismissible'];
+		$notice_wrapper_extra = '';
+		if ( $is_dismissible ) {
+			wp_enqueue_script( 'job_manager_notice_dismiss' );
+			$notice_class        .= ' is-dismissible';
+			$notice_wrapper_extra = sprintf( ' data-dismiss-action="wpjm_dismiss_notice" data-dismiss-notice="%1$s" data-dismiss-nonce="%2$s"', esc_attr( $notice_id ), esc_attr( wp_create_nonce( self::DISMISS_NOTICE_NONCE_ACTION ) ) );
+		}
+
+		// TODO Remove hard-coded 'is-dismissable' CSS class once all notices are converted to use this class.
+		echo '<div class="notice wpjm-admin-notice wpjm-admin-notice--info is-dismissible' . esc_attr( $notice_class ) . '"' . esc_attr__( $notice_wrapper_extra ). '>';
 
 		echo '<div class="wpjm-admin-notice__top">';
-		// TODO Add icon.
+
+		// TODO Implement get_icon method.
+		if ( ! empty( $notice['icon'] ) ) {
+			// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped -- Dynamic parts escaped in the function.
+			echo get_icon( $notice['icon'], 'wpjm-notice__icon' );
```

Comment:
Not sure I understand this. What is this `get_icon` method? Is a method that does not exist?
We need to implement this. I am happy if we initially only add a method like:

```
private function get_icon( $icon_name ) {
    switch ($icon_name) {
        case 'wpjm':
            return '';// path to the actual image
    }
    return ''; // default image (maybe the wpjm one?)
}
```

We can talk about this if you want over slack.

Reply:
If you think this will take a lot of time, we can just leave the todo (without the non-existing method) and create a separate ticket to implement this.

Reply:
@aaronfc Let's chat about this, as thinking over it, I'm not sure that a `get_icon` method is required, but let's chat.

Reply:
Talked over slack and we will implement a private method for this and add the new icon asset.

Reply:
@aaronfc I've committed for this in https://github.com/Automattic/WP-Job-Manager/pull/2401/commits/722258ccd6c7dc40366f9fcafaf1e200c343d0c0. I've kept the WPJM as a fallback icon for branding/recognisability. We can always return an empty string instead if we thought against that.